Author: Otto Kekäläinen <otto@debian.org>
Date:   Fri Jul 9 09:46:00 2016 +0300
  Partially revert upstreams s390x fixes that broke ppc64el builds:
  b3f4cf7c1325ff1ca771a44ae25616d80605e8e7 for xtradb and
  02043d7c53857f598a3183c7607587bfdc46ed74 for innodb

--- a/storage/innobase/include/os0sync.h
+++ b/storage/innobase/include/os0sync.h
@@ -466,7 +466,7 @@ amount to decrement. */
 # define os_atomic_decrement_uint64(ptr, amount) \
 	os_atomic_decrement(ptr, amount)
 
-# if defined(HAVE_ATOMIC_BUILTINS)
+# if defined(IB_STRONG_MEMORY_MODEL)
 
 /** Do an atomic test and set.
 @param[in,out]	ptr		Memory location to set to non-zero
--- a/storage/xtradb/include/os0sync.h
+++ b/storage/xtradb/include/os0sync.h
@@ -517,7 +517,7 @@ amount to decrement. */
 # define os_atomic_decrement_uint64(ptr, amount) \
 	os_atomic_decrement(ptr, amount)
 
-# if defined(HAVE_ATOMIC_BUILTINS)
+# if defined(IB_STRONG_MEMORY_MODEL)
 
 /** Do an atomic test and set.
 @param[in,out]	ptr		Memory location to set to non-zero
